sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -289,15 +287,28 @@ class HadoopTableReader(
   }
 
   /**
-   * Creates a HadoopRDD based on the broadcasted HiveConf and other job properties that will be
+   * The entry of creating a RDD.
+   */
+  private def createHadoopRDD(
+      inputClassName: String, localTableDesc: TableDesc, inputPathStr: String): RDD[Writable] = {
+    if (classOf[org.apache.hadoop.mapreduce.InputFormat[_, _]]
```

Comment:
I don't know if you need a method and this type of check.. just pattern-match on the result of `hiveTable.getInputFormatClass`

Reply:
Probably, @srowen meant you'd be better to just write...;
```
      // Create local references so that the outer object isn't serialized.
      val localTableDesc = tableDesc
      partDesc.getInputFileFormatClass match {
        case c: Class[_] if c == classOf[org.apache.hadoop.mapreduce.InputFormat[_, _]] =>
          createNewHadoopRdd(...)
        case _ =>
          createOldadoopRdd(...)
      }
      ...
```

Reply:
Yeah, it's not quite as simple as I thought as we have a Class not an instance, but this works if you use .isAssignableFrom

Reply:
I get it, done!